lib/src/range/range.dart

```diff
@@ -185,13 +191,17 @@ class Range<T extends Comparable<T>> {
   bool get upperBounded => upper != null;
 
   /// Whether `this` [contains] any values.
-  bool get isEmpty  => _boundValuesEqual && !isClosed;
+  bool get isEmpty => _boundValuesEqual && !isClosed;
 
   /// Whether `this` [contains] exactly one value.
   bool get isSingleton => _boundValuesEqual && isClosed;
 
-  bool get _boundValuesEqual =>
-      bounded && Comparable.compare(lower, upper) == 0;
+  bool get _boundValuesEqual {
+    final l = lower;
+    final u = upper;
```

Comment:
@hovadur could you please remove the declarations above (here and in other places where you added similar declarations) and add `!` operator to `lower` and `upper`:

```dart
if (lower == null || upper == null) return false;
return bounded && Comparable.compare(lower!, upper!) == 0;
```

Reply:
done